Board.java

```diff
@@ -51,9 +51,9 @@ public boolean checkForWin(char player) {
 
     public ArrayList availablePositions() {
         ArrayList<Integer> positions = new ArrayList<>();
-        for (int i=0; i < dimension; i++) {
+        for (int i=1; i < dimension; i++) {
```

Comment:
if i starts at 1 then the range is one-indexed and the range is 1-9.  If dimension == 9 then the terminating condition _i < dimension_ is now incorrect.  
Put in a test for this!